### DISCUSSION TITLE
Exploring Server-Driven UI with Redwood: Pros, Cons, and Impact on iOS

### DISCUSSION THREAD

**Discussion by kmithi1** [MobileNativeFoundation/discussions#219] [General]
Hello everyone, 

I've developed a server-driven UI framework where UI templates are defined in JSON and widgets/sections are coded natively for each platform (Android and iOS). While current framework is functional, developers are not happy with it as they need to work on JSON than code, its hard to debug(no step breakpoint), expression(a+b-c) are not verbose.

I'm considering switching to Redwood, which utilizes Jetpack Compose to allow updates to UI and business logic over the wire. If you've used Redwood, I'd appreciate your insights: 

- How has your experience been with this tool? 
- What are the major challenges or limitations you've encountered? 
- How has the transition affected your iOS team, especially with the gradual shift to Compose? 
- Is managing and maintaining of bundles straightforward? 
- How do you implement hardware related feature like location, camera, BLE etc.
 
Looking forward to your thoughts and experiences. Thank you!

PS: I have also checked thread on [Server-driven UI (or Backend driven UI) strategies](https://github.com/MobileNativeFoundation/discussions/discussions/47?sort=top) but Redwood discussion is missing.

**Comment by swankjesse** [comment]
I work on Redwood at Cash App. I’ll use the name ‘Treehouse’ here, which is Redwood (view system) plus Zipline (dynamic code system).

Our initial roll out was difficult – we needed to optimize a bunch of things! But I believe we’ve got something that works nicely for us.

Some Treehouse benefits:
 - It’s self-contained. We’d like adopting Treehouse to feel like adopting Lottie: it’s a new kind of view and that’s it! It doesn’t ask you to throw out everything you’ve already built.
 - It’s built on Kotlin, Compose and coroutines. You don’t need to train your team on JavaScript.
 - It has a great mechanism to integrate your existing mobile APIs.

Treehouse isn’t (yet) a packaged product. Getting ‘hello world’ running in your app for the first time is way too much work. We’re pre-1.0 and make breaking API changes. If you’d like to experiment with it, please reach out on [the Kotlin Slack](https://slack-chats.kotlinlang.org/c/squarelibraries).
